jetty-util/src/main/java/org/eclipse/jetty/util/compression/CompressionPool.java

```diff
@@ -56,73 +65,102 @@ public CompressionPool(int capacity)
     /**
      * @return Object taken from the pool if it is not empty or a newly created Object
      */
-    public T acquire()
+    public Entry acquire()
     {
-        T object;
-
-        if (_capacity == 0)
-            object = newObject();
-        else
+        Entry entry = null;
+        if (_pool != null)
         {
-            object = _pool.poll();
-            if (object == null)
-                object = newObject();
-            else if (_capacity > 0)
-                _numObjects.decrementAndGet();
+            Pool<Entry>.Entry acquiredEntry = _pool.acquire(e -> new Entry(newObject()));
+            if (acquiredEntry != null)
+            {
+                entry = acquiredEntry.getPooled();
+                entry.setEntry(acquiredEntry);
+            }
         }
 
-        return object;
+        return (entry == null) ? new Entry(newObject()) : entry;
     }
 
     /**
-     * @param object returns this Object to the pool or calls {@link #end(Object)} if the pool is full.
+     * @param entry returns this Object to the pool or calls {@link #end(Object)} if the pool is full.
      */
-    public void release(T object)
+    public void release(Entry entry)
     {
-        if (object == null)
-            return;
+        entry.release();
+    }
 
-        if (_capacity == 0 || !isRunning())
+    @Override
+    protected void doStart() throws Exception
+    {
+        if (_capacity > 0)
+            _pool = new Pool<>(Pool.StrategyType.RANDOM, _capacity, true);
```

Comment:
I think this will make a very large array list if capacity is MAX_VALUE?!?!?!?
I think by default we want to pool to be 1024 or similar

Reply:
`INFINITE_CAPACITY` is probably not an option and should be renamed to `DEFAULT_CAPACITY` or similar

Reply:
I assume that the pool wouldn't pre-allocate the max capacity. But I'm fine with changing it to `DEFAULT_CAPACITY=1024`.